### ISSUE TITLE
 Is consuming lots of ram ( Dota 2 GNU/Linux)

### ISSUE THREAD

**Issue by zezaocapoeira** [ValveSoftware/Dota-2#1417]
Hello.

(sorry bad english )

After the update of 12/03/2018 the Dota-2 (GNU / Linux version) is consuming
lots of ram.
![captura de tela_2018-03-12_17-35-12](https://user-images.githubusercontent.com/13563840/37308848-f4c8be10-261d-11e8-830d-132769d3937d.png)

System description:

https://pastebin.com/raw/HHjwLsZ7

Before the update of 12/03/2018 , it was normal using approximately 2 gb in game.

Thanks for listening.

**Comment by ywaby** [comment]
same

**Comment by sylviinho** [comment]
crashing sporadically on mac after the update

**Comment by kisak-valve** [comment]
Hello @sylviinho, please open a new issue report so that your issue can be tracked properly.

**Comment by vonclutch** [comment]
i think @kisak-valve it's related.

when the game start after selection phase, when the rune spawn ( at 00:00) when there is a big fight or just randomly (maybe cause of some spell), your game can crash because of too much memory consumtion. And most of the time it's not only the game, my 2 last game since this patch crash my computer ( and many times my game ) and now i can't really play in case i got abandon and 6month.

**Comment by sylviinho** [comment]
@kisak-valve I really think it's related too. It's not only that the game crashes, but it runs VERY slowly whenever multiple actions are being made, thus making me strongly believe the game crashes because of the RAM usage

**Comment by gdrewb-valve** [comment]
So far we haven't seen memory usage anywhere near that high at Valve so we don't have any leads.

**Comment by zezaocapoeira** [comment]
Hello.

( sorry bad english )

In this build today ( version 2,745 )  03/13/2018, apparently corrected the ram problem I had quoted.

![captura de tela_2018-03-13_22-43-29](https://user-images.githubusercontent.com/13563840/37379136-faf4a1ca-2710-11e8-8806-c573363287de.png)


Thanks for listening.

**Comment by gdrewb-valve** [comment]
Thanks.  Is anybody else still seeing memory problems?

**Comment by ywaby** [comment]
still happen, speical in imba mode, dota2 easily eat all my memery.

**Comment by AlejandroC90** [comment]
Same here in Arch Linux, Dota 2 blocks my pc because it uses 4GB of RAM (of total 6GB), sometimes happens slow and sometimes almost instantly after spectating a game

**Comment by gdrewb-valve** [comment]
@ywaby and @AlejandroC90, this memory usage only started with the Dota Plus update?

**Comment by ywaby** [comment]
yes

**Comment by cuteredpwnda** [comment]
Having the same issue here, also using Arch (Antergos) with 8GB of RAM. Game randomly crashes and kinda locks my PC after playing with low fps for a while.

**Comment by vonclutch** [comment]
Same for me try with bot in low fps and game crash once and next time my computer freeze and need to force restart

(Intel Core i7-4770 @ 3.40GHz / RAM:  7921 Mb )

**Comment by AlejandroC90** [comment]
Yes, sometimes it happens pretty fast, this was just after a turbo game
![image](https://user-images.githubusercontent.com/5132907/37436463-0d9fde20-27b6-11e8-95bf-4b5733298e5c.png)
Full RAM and full SWAP used

**Comment by cuteredpwnda** [comment]
Oh, I forgot to mention my specs/setup:
Ryzen5 1600@3.2Ghz, 8GB RAM and Gnome3 as DE.

**Comment by conrev** [comment]
I also can confirm this issue on Ubuntu 16.04 LTS (i7 7700HQ/8GB/1050ti/Unity)

Dota 2 (Vulkan) consumes around 6-7GB after 1 game. Sometimes i can even reproduce this by only watching replays.

I will test the oGL version and report here.
EDIT: I tested the OpenGL ver. and it stays at around ~2.7GB,  so maybe it is vulkan specific? (at least on my machine)

**Comment by gdrewb-valve** [comment]
Did the March 20 update improve things for any Linux users with issues?

**Comment by zezaocapoeira** [comment]
Hello.

( sorry bad english )

This build today:
![build](https://user-images.githubusercontent.com/13563840/37684967-8f7fcbe4-2c70-11e8-9224-5778d50954ff.png)

I did a screencast of a fast gameplay  with vulkan enabled , everything is working normally:

[Screencast Dota 2 ](https://youtu.be/piEzEurxhIc)

Thanks for listening.

**Comment by AlejandroC90** [comment]
Updated, watched some games, and stills uses all the ram that it can 
![photo_2018-03-20_17-14-59](https://user-images.githubusercontent.com/5132907/37685671-427fbae2-2c62-11e8-9a9d-848f711b964e.jpg)

After closing the game 
![photo_2018-03-20_17-15-02](https://user-images.githubusercontent.com/5132907/37685687-4ac78a40-2c62-11e8-9360-99ccc44dbd47.jpg)

![image](https://user-images.githubusercontent.com/5132907/37685712-71603ac6-2c62-11e8-9932-52380e037bf3.png)

**Comment by ywaby** [comment]
issue had fixed，thanks for you good job.

**Comment by vonclutch** [comment]
Looks like nothing bug anymore :+1:

**Comment by gdrewb-valve** [comment]
Anybody else still seeing problems?

**Comment by gdrewb-valve** [comment]
Closing as nobody seems to be having issues.

**Comment by jgillich** [comment]
[I still have this problem](https://user-images.githubusercontent.com/347965/40546113-93370d6e-602e-11e8-94bb-9faafb59ac1a.jpg). Game starts at around 2GB, but RAM usage constantly climbs as the game goes on. In some games I don't notice it at all, in others either Dota crashes or it hangs (due to swapping but still constantly increasing ram usage, FPS drops to zero). I've read that it might be related to cut down trees, so I've jumped into a practice lobby and cut down lots of them, it uses a whopping 5GB of RAM only a couple of minutes into the game. This is using the Vulkan API and lowest settings. I repeated the test with OpenGL and it appears to be stable at only 1.2GB RAM usage.
